runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkPipelineRunner.java

```diff
@@ -98,6 +100,8 @@ PortablePipelineResult runPipelineWithTranslator(
             ? trimmedPipeline
             : GreedyPipelineFuser.fuse(trimmedPipeline).toPipeline();
 
+    fusedPipeline = lengthPrefixKeyCoderOfStatefulStages(fusedPipeline);
```

Comment:
The operations above are already repeated in `SparkPipelineRunner`

Reply:
The sequence of steps (trim, fuse, and now lengthPrefixKeyCoderOfStatefulStages) that repeats between Flink and Spark could be moved to a utility in `org.apache.beam.runners.core.construction.graph`

Reply:
And could potentially be pulled up into `JobInvocation`, which is already shared.

Reply:
Just saw your comment when I updated the PR. I've moved the logic for key length prefixing to `GreedyStageFuser`. Good idea to move the other functions into a utility class, but I believe this is out of scope for this PR.